src/components/user-group/uc/deleteUserData.unit.test.js

```diff
@@ -0,0 +1,91 @@
+const sinon = require('sinon');
+const { ObjectId } = require('mongoose').Types;
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const deleteUserData = require('./deleteUserData.uc');
+
+const { classesRepo } = require('../repo/index');
+const { ValidationError } = require('../../../errors');
+const { toString: idToString } = require('../../../helper/compare').ObjectId;
+
+const { expect } = chai;
+chai.use(chaiAsPromised);
+
+const USER_ID = new ObjectId();
+const USER_ID_WITH_NO_CLASS = new ObjectId();
+const CLASS_ID = new ObjectId();
+const createTestGetClassForUserResult = (classId, student, teacher) => {
+	if (!classId) {
+		return [];
+	}
+
+	return [
+		{
+			_id: classId,
+			student,
+			teacher,
+		},
+	];
+};
+
+const initStubs = ({ classId, isStudent, isTeacher }) => {
+	const getClassesStub = sinon.stub(classesRepo, 'getClassesForUser');
+	getClassesStub.callsFake(() => []);
+	getClassesStub.withArgs(USER_ID).returns(createTestGetClassForUserResult(classId, isStudent, isTeacher));
+
+	const removeClassesStub = sinon.stub(classesRepo, 'removeUserFromClasses');
+	removeClassesStub.withArgs(USER_ID).returns({ n: 1, nModified: 1 });
+
+	return { getClassesStub, removeClassesStub };
+};
+
+describe('delete class user data usecase', () => {
+	describe('delete class user data uc', () => {
+		it('should return a function', async () => {
+			const deleteUserDataFromClasses = deleteUserData.deleteUserData();
+			expect(deleteUserDataFromClasses).to.be.an('Array');
+			expect(deleteUserDataFromClasses.length).to.be.equal(1);
+			expect(deleteUserDataFromClasses[0]).to.be.a('function');
+		});
+
+		it('should return a valid result (trashbin) object', async () => {
+			const { getClassesStub, removeClassesStub } = initStubs({ classId: CLASS_ID, isStudent: true, isTeacher: true });
```

Comment:
Using sinon in this way to test unit test ..looks good. Maybe we can some of this to the best practice.